### PR TITLE
fix(common): scan images once page is loaded

### PR DESCRIPTION
**Pull request by arturovt**
This commit updates the implementation of the `ImagePerformanceWarning` and runs the image scan even if the page has already been loaded. The `window.load` event would never fire if the page has already been loaded; that's why we're checking for the document's ready state.

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
// cc @atcastle

File: packages/core/src/image_performance_warning.ts [hunk 1]
```diff
@@ -49,7 +49,16 @@ export class ImagePerformanceWarning implements OnDestroy {
       // Angular doesn't have to run change detection whenever any asynchronous tasks are invoked in
       // the scope of this functionality.
       this.ngZone.runOutsideAngular(() => {
-        this.window?.addEventListener('load', waitToScan, {once: true});
+        // Consider the case when the application is created and destroyed multiple times.
```
Comment (atcastle) [inline packages/core/src/image_performance_warning.ts:52]: nit: I think this comment is a little more detailed than necessary. The `doc.readyState` check is pretty understandable as is.

**Review by atcastle — APPROVED** [review]
LGTM

File: packages/core/src/image_performance_warning.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/core/src/image_performance_warning.ts:52]: I'd propose keeping the comment, it provides extra info on the type of use-cases that we didn't cover with the `onload` event.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 7affa5775427e92ef6e949c879765b7c8aa172da.

**Pull request closed without merging**